### PR TITLE
Create asgard namespace

### PR DESCRIPTION
**Pull request by quantumsteve**
Please review the [developer documentation](https://github.com/project-asgard/asgard/wiki/developing)
on the wiki of this project that contains help and requirements.

## Proposed changes

Describe what this PR changes and why.  If it closes an issue, link to it here
with [a supported keyword](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

This adds the asgard namespace to all library files. 

## What type(s) of changes does this code introduce?
_Put an `x` in the boxes that apply._

- [ ] Bugfix
- [x] New feature
- [x] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

### Does this introduce a breaking change?

- [x] Yes
- [ ] No

## Checklist

_Put an x in the boxes that apply. You can also fill these out after creating
the PR. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- [x] this PR is up to date with current the current state of 'develop'
- [x] code added or changed in the PR has been clang-formatted
- [ ] this PR adds tests to cover any new code, or to catch a bug that is being fixed
- [ ] documentation has been added (if appropriate)

### CODE REVIEW

File: src/boundary_conditions.hpp [hunk 1]
```diff
@@ -6,10 +6,11 @@
 #include "transformations.hpp"
 
 template<typename P>
-using unscaled_bc_parts = std::vector<std::vector<std::vector<fk::vector<P>>>>;
+using unscaled_bc_parts =
```
Comment (mkstoyanov) [inline src/boundary_conditions.hpp:12]: This should also be inside a namespace, right?

File: src/lib_dispatch.cpp [hunk 2]
```diff
@@ -84,6 +84,8 @@ extern "C"
 
 #endif
 
+namespace
```
Comment (mkstoyanov) [inline src/lib_dispatch.cpp:89]: Should this be `namespace lib_dispatch`

What's the namespace, I didn't even think this is valid C++.

**Review by mkstoyanov — CHANGES_REQUESTED** [review]
Better, actually we can have a library now.

File: src/lib_dispatch.cpp (hunk 2 above)
Comment (quantumsteve) [reply src/lib_dispatch.cpp:89]: unnamed namespaces are valid C++

https://abseil.io/tips/186

Comment (mkstoyanov) [reply src/lib_dispatch.cpp:89]: Yes, I learned something new :) 

However, why do we need an unnamed one here? Shouldn't this be part of `lib_dispatch` namespace?

Comment (quantumsteve) [reply src/lib_dispatch.cpp:89]: The `device_handler` struct and `cublas_trans` function are only used inside lib_dispatch.cpp. Putting them inside an unnamed namespace means it's local to this translation unit and if another symbol with the same name is defined elsewhere there will not be a violation of the One Definition Rule.

File: src/boundary_conditions.hpp (hunk 1 above)
Comment (quantumsteve) [reply src/boundary_conditions.hpp:12]: I moved `unscaled_bc_parts` into `asgard::boundary_conditions`.

**Review by ckendrick — APPROVED** [review]

File: src/lib_dispatch.cpp (hunk 2 above)
Comment (mkstoyanov) [reply src/lib_dispatch.cpp:89]: Ok, that makes sense.

**Review by mkstoyanov — APPROVED** [review]

**Comment by mkstoyanov** [comment]
@quantumsteve merge when you want

**Pull request merged**